docs/madr/decisions/056-meshexternalservice-and-zoneegress.md

```diff
@@ -0,0 +1,406 @@
+
+##### MeshCircuitBreaker
+
+###### Old ExternalService behaviour
+
+Circuit Breaker is configured on the sidecar always.
+
+###### New behaviour
+
+If on egress
+- services from one zone will act as an aggregate - if a CB triggers by service-1 then service-2 will also see that result immediately
+  - this makes the configuration more sensitive to drastic changes
+  - if a MES starts failing on one endpoint /example1 that is only consumed by service-1 it will also trigger for service-2 that might be happily consuming endpoint /example2
+  - this could be mitigated by re-defining a MeshExternalService under a different name (a bit awkward)
+
+###### Verdict
+
+There might be situations where it makes sense on egress but seems like it's more natural on the sidecar.
+
+##### MeshFaultInjection
+
+###### Old ExternalService behaviour
+
+FaultInjection and MeshFaultInjection is configured for ExternalService on egress inbound.
+
+###### New behaviour
+
+If on sidecar:
+- less traffic in the cluster (it won't even exit the pod)
+- you won't be able to validate if your policies (MeshRetry, MeshTimeout, etc.) work correctly because they won't be triggered
+
+###### Verdict
+
+Probably should stay on egress.
+
+##### MeshHealthCheck
+
+###### Old ExternalService behaviour
+
+HealthCheck is configured on the sidecar.
+
+###### New behaviour
+
+If on egress:
+- Seems to be the same story as circuit breaker in terms of acting as an aggregate / sensitivity
+- Doing it on the egress could cause less traffic to MES (assuming there is fewer instances of egress than services HC-ing)
+
+###### Verdict
+
+To me, it makes a little bit more sense (than circuit breaker) to have HC be configurable on where it's placed
+
+##### MeshMetric
+
+It's a new policy without a direct ancestor.
+MeshMetric is not applied on egress at all, and we don't provide a way to configure it.
+I don't see any reason not to allow this in the future.
+
+##### MeshProxyPatch
+
+###### Old ExternalService behaviour
+
+It does not look like it's even possible to modify egress with MeshProxyPatch or ProxyTemplate so this needs work outside the scope of this MADR,
+but it makes sense to be able to do this.
+
+##### MeshRateLimit
+
+###### Old ExternalService behaviour
+
+Both RateLimit and MeshRateLimit are configured on the egress.
+
+###### New behaviour
+
+If on sidecar:
+- limit is multiplied by the number of instances (which is bad)
+- less traffic in the mesh (traffic does )
+
+###### Verdict
+
+It seems that it only makes sense on egress, we don't want the ratelimit to be changing with number of instances.
+
+##### MeshRetry
+
+###### Old ExternalService behaviour
+
+Both Retry and MeshRetry are configured on the sidecar
+
+###### New behaviour
+
+If on egress:
+- when an ExternalService is unavailable a retry will only occur `numRetries` tries - not `numRetries` * instances (in sidecar case) - which is good because we're not overwhelming the service
```

Comment:
> - when an ExternalService is unavailable a retry will only occur `numRetries` tries - not `numRetries` * instances

Is this true? I can't quite find where this s documented/implied in the docs. Aren't retries tracked per request?

Reply:
> Is this true? I can't quite find where this s documented/implied in the docs. Aren't retries tracked per request?

I think you're right. I confused that with HC / outlier detection which will behave differently